### ISSUE TITLE
tutorial please!

### ISSUE THREAD

**Issue by kieran-mace** [julipetal/petalNet#3]
Not sure how to get started with this. Can you make a minimal script that steps through a regular workflow?

**Comment by nishagove** [comment]
Yeah. Looking forward to a tutorial. I find it rather confusing especially when the file labels of each function don't match with the other. For example, the function 1 generates an output that would serve as an input for function 2. I get lost when the names differ. A tutorial will certainly resolve all confusions.

**Comment by julipetal** [comment]
I am sorry for the delay - I needed a break from the computer and drove across country and disconnected. 

You only really have to call one function 
`dataToVNs(dataFile, GoIFile, annoFile, metric, thresholds, MMoExists)`

`dataFile `is your gene expression file in tab-separated .txt format, rows are individual genes and columns are conditions/measurements
`GoIFile `is another tab (or line)-separated .txt file including genes that are of interest to you

_all other inputs are optional_
`annoFile `stands for annotation file. If you have an annotation file for your genes (or whatever your node definition is), this file can be uploaded as well and then the groups (vicinity networks) will be annotated. This file can include anything you want, it just needs to be a  tab-separated .txt file and the first column needs to include (not all or more) gene identifies matching the ones from the expression input file
`metric `is the association measure used to define edges between nodes, default is Spearman. The association measure can be set to other correlation or distance measures, if you are working with RNAseq - I recommend leaving it as is
`thresholds `is a numeric vector indicating where to cut of the weight for the edges, I recommend not to specify values, the program will calculate the best thresholds
`MMoExists `can be ignored 

so your function call could look something like this
`dataToVNs("myDataFile.txt", "myGenesOfInterest.txt")`
OR
`dataToVNs("myDataFile.txt", "myGenesOfInterest.txt", "myAnnotationFile.txt")`

**Comment by nishagove** [comment]
The function stands on its own. Initially,  I thought that one is dependent on the other. For instance I used the readinExpM function to convert my txt. file into data matrix , later fed into graphHistQQ function. Short cut would be to read my txt. file in the graphHistQQFromFile function, which does the conversion and eventually gives the output(histogram).May I know where the output(histogram) is stored and how to retrive them?
Normally, the png file will just pop out. I cant allocate them.These were the codes used;

inputFile=read.table("countdata.txt")
> graphHistQQFromFile <- function(inputFile, fileName="Data_Hist-QQ"){
+ 
+ 	m1 = scan(inputFile, sep="\t", what="", nlines=1)
+ 	m2 = scan(inputFile, sep="\t", what="", nlines=1, skip=1)
+ 
+ 	if(isTRUE(length(m1)==length(m2))){
+ 		expression     = matrix(scan(inputFile,what="",sep="\t", skip=1),ncol=length(m1), byrow=TRUE)
+ 		expM           = matrix(as.numeric(expression[,-1]), ncol = length(m1)-1, byrow=FALSE)
+ 		colnames(expM) = m1[-1]
+ 		rownames(expM) = expression[,1]
+ 	}else{
+ 		expression     = matrix(scan(inputFile, sep="\t", what="", skip=1), ncol=length(m2), byrow=TRUE)
+ 		expM           = matrix(as.numeric(expression[,-1]), ncol=length(m2)-1, byrow=FALSE)
+ 		colnames(expM) = m1
+ 		rownames(expM) = expression[,1]
+ 	}
+ 	rm(m1, m2,expression)
+ 
+ 	save(expM, file="expM.RData")
+ 	
+ 	measures = as.vector(expM)
+ 	minExp   = round(min(measures, na.rm=TRUE)-0.5,digits=0)
+ 	maxExp   = round(max(measures, na.rm=TRUE)+0.5, digits=0)
+ 	delta    = (maxExp -minExp)/100
+ 
+ 	numObs = length(measures)
+ 	mitte  = mean(measures, na.rm=TRUE)
+ 	sdev   = sd(measures, na.rm = TRUE)
+ 
+ 	png(filename=paste(fileName,".png", sep=""), width=7, height=8.5, units="in", pointsize=12, bg="white", res=600)
+ 	par(mfrow = c(2,1))
+ 	h = hist(measures, breaks=seq(minExp,maxExp , by=delta), plot=FALSE)
+ 
+ 	xfit = seq(min(measures, na.rm=TRUE),max(measures, na.rm=TRUE),length=100)
+ 	yfit = dnorm(xfit,mean=mean(measures, na.rm=TRUE),sd=sd(measures, na.rm=TRUE))
+ 	yfit = yfit*diff(h$mids[1:2])*length(measures)
+ 
+ 	plotMaxV = max(c(h$counts,yfit))
+ 	plot(h, ylim=c(0,plotMaxV), main="Histogram of Data", col="blue")
+ 	lines(xfit, yfit, col="red", lwd=2) 
+ 	
+ 	qqnorm(expM)
+ 	qqline(expM, col="red")
+ 	dev.off()
+ }	
>
